### PR TITLE
Add client connection code sample in readme

### PR DESCRIPTION
**Pull request by jayeshathila**
Signed-off-by: jayeshathila <sharma.jayesh52@gmail.com>

### Description
Add client connection code sample in readme
  
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by VijayanB** [comment]
Will you be adding opensearch-dsl in another PR?

**Comment by jayeshathila** [comment]
> Will you be adding opensearch-dsl in another PR?

We already have dsl examples in `opensearch-dsl` module - https://github.com/opensearch-project/opensearch-ruby/blob/main/opensearch-dsl/README.md . Those should suffice IMO.

File: README.md [hunk 1]
```diff
@@ -12,6 +12,107 @@ OpenSearch Ruby Client
...
+
+## Sample code
+
+```
+require 'opensearch'
+
+client = Opensearch::Client.new url: 'http://localhost:9200', log: true
+
+# If you want to use authentication credentials
+# client = Opensearch::Client.new url: 'http://USERNAME:PASSWORD@myhost:8080', log: true
```
Comment (VijayanB) [inline README.md:36]: Shall we use default opensearch distribution ( with security ) for example
```suggestion
client = Opensearch::Client.new url: 'https://admin:admin@localhost:9200', log: true
```

Comment (jayeshathila) [reply README.md]: Fixed

**Comment by jayeshathila** [comment]
> > Will you be adding opensearch-dsl in another PR?
> 
> We already have dsl examples in `opensearch-dsl` module - https://github.com/opensearch-project/opensearch-ruby/blob/main/opensearch-dsl/README.md . Those should suffice IMO.

Added

File: README.md [hunk 2]
```diff
@@ -12,6 +12,109 @@ OpenSearch Ruby Client
...
+If you prefer to add the client manually or just want to examine the source code, see [opensearch-ruby on GitHub](https://github.com/opensearch-project/opensearch-ruby).
+
+## Sample code
+
+```
+require 'opensearch'
+
+
+# If you want to use authentication credentials
+client = Opensearch::Client.new url: 'https://admin:admin@localhost:9200', log: true
```
Comment (VijayanB) [inline README.md:35]:
```suggestion
client = OpenSearch::Client.new url: 'https://admin:admin@localhost:9200', log: true
```

File: README.md [hunk 3]
```diff
@@ -12,6 +12,109 @@ OpenSearch Ruby Client
...
+
+# If you don't want to use authentication credentials
+#client = Opensearch::Client.new url: 'http://localhost:9200', log: true
```
Comment (VijayanB) [inline README.md:38]:
```suggestion
#client = OpenSearch::Client.new url: 'http://localhost:9200', log: true
```

File: README.md [hunk 4]
```diff
@@ -12,6 +12,109 @@ OpenSearch Ruby Client
 
 **opensearch-ruby** is [a community-driven, open source fork](https://aws.amazon.com/blogs/opensource/introducing-opensearch/) of elasticsearch-ruby licensed under the [Apache v2.0 License](LICENSE.txt).
 For more information, see [opensearch.org](https://opensearch.org/).
+
+## Setup
+
+To add the client to your project, install it using [RubyGem](https://rubygems.org/)
+
+`gem install opensearch`
```
Comment (VijayanB) [inline README.md:20]: or install it from a source code checkout:

```
git clone https://github.com/elasticsearch/opensearch-ruby.git
cd opensearch-ruby/opensearch
gem build opensearch.gemspec
gem install opensearch-<version>.gem
```

Comment (jayeshathila) [reply README.md:20]: Done

File: README.md (hunk 3 above)
Comment (jayeshathila) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (jayeshathila) [reply README.md]: Fixed

**Review by VijayanB — APPROVED** [review]

**Review by vamshin — APPROVED** [review]
LGTM! Thanks

**Pull request merged**